### DISCUSSION TITLE
Script to increment build number fails

### DISCUSSION THREAD

**Discussion by MbuthiaWaKihara** [codemagic-ci-cd/codemagic-docs#2864] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

React Native

### Steps to reproduce

All the build steps work fine, but the publishing step fails. This is because the build number already exists. 
I have a script to increment the build number. Here's the script:

`           LATEST_GOOGLE_PLAY_BUILD_NUMBER=$(google-play get-latest-build-number --package-name "$PACKAGE_NAME")
            echo "LATEST_GOOGLE_PLAY_BUILD_NUMBER: $LATEST_GOOGLE_PLAY_BUILD_NUMBER"
            if [ -z $LATEST_GOOGLE_PLAY_BUILD_NUMBER ]; then
              # fallback in case no build number was found from Google Play.
              # Alternatively, you can `exit 1` to fail the build
              # BUILD_NUMBER is a Codemagic built-in variable tracking the number
              # of times this workflow has been built
                UPDATED_BUILD_NUMBER=$BUILD_NUMBER
            else
                UPDATED_BUILD_NUMBER=$(($LATEST_GOOGLE_PLAY_BUILD_NUMBER + 1))
            fi
`

The variable UPDATED_BUILD_NUMBER is assigned the value BUILD_NUMBER, because getting the latest build number fails.
I also see this in the logs:
`google-play: error: argument --credentials: Provided value "********" is not valid`

I am not sure why this happens because I have all the credentials set up fine, the build is also generated fine. What could I be missing?

### Expected results

Correct latest build number value is fetched from PlayStore

### Actual results

Fallback build number value is used

### Build id (optional)

674ea9be04baa359747fab4a

**Comment by dam-ease** [comment, marked as answer]
Hello @MbuthiaWaKihara,
Please can you confirm that the value of `GCLOUD_SERVICE_ACCOUNT_CREDENTIALS` is in plain text and not base64 encoded? If you would like to encode the value, then you will need to decode it before the `google-play get-latest-build-number` step using
```
export GCLOUD_SERVICE_ACCOUNT_CREDENTIALS=$(echo $GCLOUD_SERVICE_ACCOUNT_CREDENTIALS | base64 --decode)
```
More details about this can also be found [here](https://github.com/codemagic-ci-cd/cli-tools/blob/master/docs/google-play/get-latest-build-number.md#optional-arguments-for-command-google-play)

**Reply by MbuthiaWaKihara** [reply]
Adding `GCLOUD_SERVICE_ACCOUNT_CREDENTIALS` as plain text fixed the issue. Thanks!
